### PR TITLE
Migrate CustomLifeCycleTestCase

### PR DESCRIPTION
**Pull request by nalakas**

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java [hunk 1]
```diff
@@ -44,53 +43,72 @@
...
         lifeCycleAdminClient.editLifeCycle(apiLifeCycleName, customizedAPILifecycleContent);
 
-        apiCreationRequestBean = new APICreationRequestBean(API_NAME, API_CONTEXT, API_VERSION_1_0_0,
-                user.getUserName(), new URL(apiEndPointUrl));
-        apiIdentifier = new APIIdentifier(user.getUserName(), API_NAME, API_VERSION_1_0_0);
-        publisherURLHttp = getPublisherURLHttp();
-        apiPublisherClient = new APIPublisherRestClient(publisherURLHttp);
-        apiPublisherClient.login(user.getUserName(), user.getPassword());
+        String gatewayUrl;
+        if (gatewayContextWrk.getContextTenant().getDomain().equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java]: Swap the equals condition.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java [hunk 2]
```diff
@@ -44,53 +43,72 @@
...
-//        APILifeCycleStateRequest apiLifeCycleStatusChangeRequest = new APILifeCycleStateRequest(API_NAME,
-//                user.getUserName(), APILifeCycleState.PROMOTE);
-//        apiLifeCycleStatusChangeRequest.setVersion(apiIdentifier.getVersion());
-//
-//        //Change api status to custom state
-//        HttpResponse publishAPIResponse = apiPublisherClient.changeAPILifeCycleStatus(apiLifeCycleStatusChangeRequest);
-//        Boolean statusChangeSuccess = verifyAPIStatusChange(publishAPIResponse, APILifeCycleState.PUBLISHED,
-//                APILifeCycleState.PROMOTED);
-//        Assert.assertTrue(statusChangeSuccess, "API status Change is not successful");
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, Constants.PUBLISHED);
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java]: Lets use APILifeCycleAction.PUBLISHED.getAction()

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java [hunk 3]
```diff
@@ -44,53 +43,72 @@
...
+        Assert.assertEquals(APILifeCycleState.PUBLISHED.getState(), restAPIPublisher.getLifecycleStatus(apiId).
+                getData(), "lifecycle not changed to published");
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, PROMOTE);
+        Assert.assertEquals(PROMOTE + "d", restAPIPublisher.getLifecycleStatus(apiId).getData(),
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java]: Why are we using it as PROMOTE + "d"? We should be able to use APILifeCycleState.PROMOTED.getState()

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java [hunk 4]
```diff
@@ -44,53 +43,72 @@
...
     }
 
     @AfterClass
     public void cleanupArtifacts() throws Exception {
         //Remove test api and revert to original lifecycle config
-        deleteAPI(apiIdentifier, apiPublisherClient);
-        lifeCycleAdminClient.editLifeCycle(apiLifeCycleName, originalLifeCycleContent);
-        loginClient.logOut();
-        apiPublisherClient.logout();
+        restAPIPublisher.deleteAPI(apiId);
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java]: Lets have the cleanUp() at the end as a practice

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java (hunk 1 above)
Comment (nalakas) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java (hunk 2 above)
Comment (nalakas) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java (hunk 3 above)
Comment (nalakas) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java (hunk 4 above)
Comment (nalakas) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java [hunk 5]
```diff
@@ -1,96 +1,106 @@
...
-import org.testng.annotations.BeforeClass;
-import org.testng.annotations.Test;
+import org.testng.annotations.*;
 import org.wso2.am.admin.clients.lifecycle.LifeCycleAdminClient;
-import org.wso2.am.integration.test.utils.bean.APICreationRequestBean;
-import org.wso2.am.integration.test.utils.bean.APILifeCycleState;
-import org.wso2.am.integration.test.utils.bean.APILifeCycleStateRequest;
-import org.wso2.am.integration.test.utils.clients.APIPublisherRestClient;
-import org.wso2.carbon.apimgt.api.model.APIIdentifier;
+import org.wso2.am.integration.test.utils.bean.*;
```
[the remark this answers is not in the document]
Comment (nalakas) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java]: fixed in 
652db6f

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java [hunk 6]
```diff
@@ -1,96 +1,106 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package org.wso2.am.integration.tests.api.lifecycle;
 
 import org.testng.Assert;
-import org.testng.annotations.AfterClass;
-import org.testng.annotations.BeforeClass;
-import org.testng.annotations.Test;
+import org.testng.annotations.*;
```
[the remark this answers is not in the document]
Comment (nalakas) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/api/lifecycle/CustomLifeCycleTestCase.java]: fixed in 
652db6f

**Pull request merged**